### PR TITLE
Add dev-container and echo app example

### LINKED ISSUE #5: Create dev container image for cpp development
Create dev container image for cpp development

### PR DESCRIPTION
**Pull request by youngbupark**
Since c++ library highly depends on the dependency library in the system, this pr is adding dev-container to standardize the dev environment and add echo_app sample to demonstrate service invocation for C++ user.

* Add dev-container for c++ dev environment
* Add step by step instruction to build client
* improve makefile
* regenerate grpc proto client
* Add Echo App sample

Closes #5 #6

### CODE REVIEW

**Comment by youngbupark** [comment]
@ooeygui ^

File: examples/echo_app/components/statestore.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: dapr.io/v1alpha1
```
Comment (amanbha) [inline examples/echo_app/components/statestore.yaml]: add examples/../components to .gitignore

File: components/pubsub.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: dapr.io/v1alpha1
```
Comment (amanbha) [inline components/pubsub.yaml]: add components to .gitignore

File: components/statestore.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: dapr.io/v1alpha1
```
Comment (amanbha) [inline components/statestore.yaml]: add components to .gitignore

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
+{
```
Comment (amanbha) [inline .vscode/settings.json]: add .vscode to .gitignore.

File: examples/echo_app/components/statestore.yaml (hunk 1 above)
Comment (youngbupark) [reply examples/echo_app/components/statestore.yaml]: added.

File: components/pubsub.yaml (hunk 2 above)
Comment (youngbupark) [reply components/pubsub.yaml]: added.

File: components/statestore.yaml (hunk 3 above)
Comment (youngbupark) [reply components/statestore.yaml]: added.

File: .vscode/settings.json (hunk 4 above)
Comment (youngbupark) [reply .vscode/settings.json]: added.

**Pull request merged**